### ISSUE TITLE
[BUG] Some Sites Don't Load Randomly

### ISSUE THREAD

**Issue by monkeylove** [Izheil/Quantum-Nox-Firefox-Dark-Full-Theme#101]
**Describe the bug**
This problem started around a month ago. When I add the dark theme and addons.css patched with the UUID replacer, some sites like reddit.com don't load randomly. That is, sometimes they don't load: there's only a blank tab with the hourglass icon. The problem continues even after I did the ff.

- reset the modem and router, and restarted the PC;
- cleared the cache and cookies;
- used safe mode;
- refreshed the brower;
- created a new profile;
- reinstalled the browser;
- uninstalled and installed the browser and created a new profile;
- set Adguard to default mode;
- used uBlock Origin instead of Adguard;
- tried the nightly version of the browser;
- used the ISP DNS or another DNS; and
- disabled one or more settings in Adguard, such as the DNS, security, and extensions.

There's no problem, though, when I load another site in another tab, or load reddit.com in another browser like Edge (also protected by Adguard).

The problem disappears when I empty the chrome directory in Firefox.

**To Reproduce**
Steps to reproduce the behavior, or a brief explanation on it:
1. Install the dark theme and addons.css in the chrome directory of Firefox.
2. Start browsing.
3. Sometimes, a site would not load. There's only a blank tab with the hour glass icon on the tab.
4. After restarting the browser in normal or safe mode, the site loads correctly most of the time.
5. When the contents of the chrome directory are removed, the problem disappears.

**Expected behavior**
Sites should load correctly, and they do only if the chrome directory is empty.

**Desktop (please complete the following information):**
 - OS: Windows 10 x64 Home SL
 - FF83, Adguard for Desktop 7.5.3

**Comment by Izheil** [comment]
Are you sure it doesnt happen without the chrome folder as well? It should happen more often after first starting the browser and opening a reddit/gmail tab.

Ive seen this starting to happen to [other people on reddit](https://www.reddit.com/r/firefox/comments/j73qhe/firefox_wont_load_reddit_on_first_try_but_all/) as well after the FF78 update, and it was related to service workers. 

Overall, there [arent many benefits on using service workers](https://www.reddit.com/r/firefox/comments/7dq2h7/is_there_any_reason_not_to_disable_service_workers/) unless you use desktop notification for webpages. (At least i havent noticed any issue with them disabled so far, and thats since FF78 update).

Im not sure how would the css files you mentioned cause any issue, since they just style pages, they cant really cause breakage... Unless Firefox loads userchrome like files in some odd way that caused service workers to cause it (which, still, i dont see how would that happen).

Another way to test it is using private window with the chrome files installed. Technically, pages cant access service workers on private mode, but userchrome-like files should still load.

If the problem persists on private mode (just for testing it), then we can try some simple css files to see if there is any of the rules in the themes causing the problem.

**Comment by monkeylove** [comment]
Thanks for the help! I didn't know where to ask, and your points are very helpful. I'll bookmark them and this thread.

I'll follow your suggestions and disable service workers (I don't use desktop notifications) and see what happens.

**Comment by Izheil** [comment]
I'll test the issue with and without service workers when I have some time and see if it's indeed related to using any CSS file on the Chrome folder.

For me, since I disabled them I haven't had any issue at least, but if it was indeed some issue with using CSS files, I'll see if there was any specific CSS rule in there, and if it's just using CSS in general, I'll try to report it to Mozilla to see if they can check it.

In either case, I found [another thread](https://www.reddit.com/r/firefox/comments/hjr6da/some_websites_dont_load_at_first_try_new_firefox/) with people who did remove the userChrome customizations and still had the problem, but I think it's worth testing if I can do something.

Anyway, the Bugzilla issue is [this one](https://bugzilla.mozilla.org/show_bug.cgi?id=1649007) if you want to check progress on Mozilla fixing it... although they closed it a while ago since it was fixed on certain release, but now it's happening to people again (the lastest posts there are from 2 weeks ago).
